src/MongoDB.Bson/BsonDefaults.cs

```diff
@@ -58,30 +60,57 @@ public static IBsonSerializer DynamicArraySerializer
         /// </summary>
         public static IBsonSerializer DynamicDocumentSerializer
         {
-            get 
+            get
             {
                 if (!__dynamicDocumentSerializerWasSet)
                 {
                     __dynamicDocumentSerializer = BsonSerializer.LookupSerializer<ExpandoObject>();
                 }
-                return __dynamicDocumentSerializer; 
+                return __dynamicDocumentSerializer;
             }
-            set 
+            set
             {
                 __dynamicDocumentSerializerWasSet = true;
-                __dynamicDocumentSerializer = value; 
+                __dynamicDocumentSerializer = value;
             }
         }
 
         /// <summary>
-        /// Gets or sets the default representation to be used in serialization of 
-        /// Guids to the database. 
-        /// <seealso cref="MongoDB.Bson.GuidRepresentation"/> 
+        /// Gets or sets the default representation to be used in serialization of
+        /// Guids to the database.
+        /// <seealso cref="MongoDB.Bson.GuidRepresentation"/>
         /// </summary>
+        [Obsolete("This property will be removed in a later release.")]
         public static GuidRepresentation GuidRepresentation
         {
-            get { return __guidRepresentation; }
-            set { __guidRepresentation = value; }
+            get
+            {
+                if (BsonDefaults.GuidRepresentationMode != GuidRepresentationMode.V2)
+                {
+                    throw new InvalidOperationException("BsonDefaults.GuidRepresentation can only be used when BsonDefaults.GuidRepresentationMode is V2.");
```

Comment:
These exceptions are helpful to an application writer that is trying to convert their application to use V3 mode. Without these exceptions it would be easy to miss places where the application is still trying to use V2 mode.

I know these exceptions helped *me* find lots of places in the driver that I missed in my first pass of converting to V3 mode.